### PR TITLE
chore: remove abuse async

### PR DESCRIPTION
**Pull request by sokel**
Remove abuse returning channels in blockchain api.
Ps: everywhere it was used synchronously.

### CODE REVIEW

File: blockchain/api.go [hunk 1]
```diff
@@ -48,18 +48,18 @@ type EventsAPI interface {
 
 type MarketAPI interface {
 	QuickBuy(ctx context.Context, key *ecdsa.PrivateKey, askId *big.Int) (*types.Transaction, error)
-	OpenDeal(ctx context.Context, key *ecdsa.PrivateKey, askID, bigID *big.Int) <-chan DealOrError
-	CloseDeal(ctx context.Context, key *ecdsa.PrivateKey, dealID *big.Int, blacklisted bool) <-chan error
+	OpenDeal(ctx context.Context, key *ecdsa.PrivateKey, askID, bigID *big.Int) DealOrError
```
[the remark this answers is not in the document]
Comment (sokel) [reply blockchain/api.go]: th, fixed

File: insonmnia/node/deals.go [hunk 2]
```diff
@@ -82,20 +82,20 @@ func (d *dealsAPI) Status(ctx context.Context, id *pb.ID) (*pb.DealInfoReply, er
 }
 
 func (d *dealsAPI) Finish(ctx context.Context, req *pb.DealFinishRequest) (*pb.Empty, error) {
-	if err := <-d.remotes.eth.Market().CloseDeal(ctx, d.remotes.key, req.GetId().Unwrap(), req.GetAddToBlacklist()); err != nil {
-		return nil, fmt.Errorf("could not close deal in blockchain: %s", err)
+	if err := d.remotes.eth.Market().CloseDeal(ctx, d.remotes.key, req.GetId().Unwrap(), req.GetAddToBlacklist()); err != nil {
+		return nil, fmt.Errorf("could not close deal in blockchain: %v", err)
```
Comment (antmat) [inline insonmnia/node/deals.go]: any reasons for replacing %s to %v?

Comment (sokel) [reply insonmnia/node/deals.go]: fixed

File: cmd/pandora/ammo_marketplace.go [hunk 3]
```diff
@@ -81,12 +81,12 @@ func (m *OrderPlaceAmmo) Type() AmmoType {
 
 func (m *OrderPlaceAmmo) Execute(ctx context.Context, ext interface{}) error {
 	mExt := ext.(*marketplaceExt)
-	orderOrError := <-mExt.market.PlaceOrder(ctx, mExt.privateKey, order())
-	if orderOrError.Err != nil {
-		return orderOrError.Err
+	orderOrError, err := mExt.market.PlaceOrder(ctx, mExt.privateKey, order())
```
Comment (antmat) [inline cmd/pandora/ammo_marketplace.go]: naming

File: insonmnia/node/master.go [hunk 4]
```diff
@@ -35,18 +35,18 @@ func (m *masterMgmtAPI) WorkersList(ctx context.Context, address *sonm.EthAddres
...
 	return &sonm.Empty{}, nil
 }
 
 func (m *masterMgmtAPI) WorkerRemove(ctx context.Context, request *sonm.WorkerRemoveRequest) (*sonm.Empty, error) {
 	ctxlog.G(m.ctx).Info("handling WorkersRemove request")
-	err := <-m.remotes.eth.Market().RemoveWorker(ctx, m.remotes.key, request.GetMaster().Unwrap(), request.GetWorker().Unwrap())
+	err := m.remotes.eth.Market().RemoveWorker(ctx, m.remotes.key, request.GetMaster().Unwrap(), request.GetWorker().Unwrap())
 	if err != nil {
-		return nil, fmt.Errorf("could not remove dependant worker from blockchain: %s", err)
+		return nil, fmt.Errorf("could not remove dependant worker from blockchain: %s", err.Error())
```
Comment (antmat) [inline insonmnia/node/master.go]: try again!

Comment (sokel) [reply insonmnia/node/master.go]: fckng vt!))

File: blockchain/api.go [hunk 5]
```diff
@@ -399,45 +375,38 @@ func (api *BasicMarketAPI) placeOrder(ctx context.Context, key *ecdsa.PrivateKey
...
-		ch <- err
-		return
+		return err
 	}
-	ch <- nil
+
+	return nil
+}
+
+func (api *BasicMarketAPI) cancelOrder(ctx context.Context, key *ecdsa.PrivateKey, id *big.Int, ch chan error) {
```
Comment (antmat) [inline blockchain/api.go]: WTF?

Comment (sokel) [reply blockchain/api.go]: th, fixed

**Pull request merged**